src/diff.rs

```diff
@@ -74,51 +75,89 @@ pub struct DiffStats {
     raw: *mut raw::git_diff_stats,
 }
 
+/// Structure describing the binary contents of a diff.
+pub struct DiffBinary<'diff> {
+    raw: *const raw::git_diff_binary,
+    _marker: marker::PhantomData<&'diff Diff<'diff>>,
+}
+
+/// The contents of one of the files in a binary diff.
+pub struct DiffBinaryFile<'diff> {
+    raw: *const raw::git_diff_binary_file,
+    _marker: marker::PhantomData<&'diff Diff<'diff>>,
```

Comment:
I *think* this is ok as-is (not having two lifetimes) because there's no need to mention the `'repo` lifetime in the implementation